alica_engine/include/engine/BasicPlan.h

```diff
@@ -25,8 +25,18 @@ class BasicPlan : private RunnableObject
     using RunnableObject::getName;
 
     void notifyAssignmentChange(const std::string& assignedEntryPoint, double oldUtility, double newUtility, size_t numberOfAgents);
+    void setAsMasterPlan() { _isMasterPlan = true; };
 
 protected:
+    void setTracing(TracingType type, std::function<std::optional<std::string>(const BasicPlan*)> customTraceContextGetter = {})
+    {
+        if (customTraceContextGetter) {
+            RunnableObject::setTracing(type, [this, &customTraceContextGetter]() { return customTraceContextGetter(this); });
```

Comment:
>&customTraceContextGetter

Aren't you binding a reference to a temp object here?

Reply:
Maybe `customTraceContextGetter = std::move(customTraceContextGetter)`

Reply:
oops, you are right, will fix it